### ISSUE TITLE
New label component

### ISSUE THREAD

**Issue by SOjaHPE** [grommet/hpe-design-system#3232]
The GLCP request and previous usage in the global header HFS referred to this as a 'label'. 'Badge' is a problematic term because it has other meaning within the ux / ui space.

**What this component does**
This component will aid our users by adding additional information which classifies an item / element, e.g. the status of an item. 
 
**Guidance**
• Formal guidance required for usage. There seems to be inconsistent uses of this ‘label’ element that is meant to help draw attention by providing a means to show state, label, organize, or categorize items. 

• Current instances looks like an active state button/control in some screens – the gray wrapper resembles the hover state of a button. These should never be interactive and users should never associate or confuse them with controls or other interactive elements. The shape should have square corners to not be confused with action buttons or tags.

• Naming of this needs to be looked into since ‘badge’ or ‘badging’ is associated with other UI elements/affordances  

• This element should convey tertiary information, maybe the state 'New', 'Verified' or health of an object or entity. Still TBD

• This should never be used to display meta data or allow it to be confused with tags or filter chips.

• It should not be used to display contextual support information such as object ID’s or account #'s

• This component should never be used to replace existing elements within defined page regions e.g. pageheader title, subtitle, description, etc. They are used as a tertiary level informational element 

• This component should never contain icons or data vis. data points, content should be simple text strings

• The color palette needs to be determined. Need to avoid conflicts with existing means to communicate state / status / health. Color alone isn't accessible to show state or status; the content of the text string will convey what user's need to know if using assistive technologies

• Affinity matters, this component should be placed immediately after or near the proper name or primary identifier of an object / entity

**Examples**: real estate listings - newly listed, pending, under contract, price reduced, foreclosure, taking back up offers, Virtual Tour, Waterfront. A 'new' service, a 'verified' user.

[Link to existing article](https://blog.prototypr.io/design-guide-classifying-items-db5b121985d) supporting the justification for this element to exist, why it is different from tags, or usual meta data objects.

See this[ link for figma GLCP Glasgow use cases](https://www.figma.com/file/8pInYUjeKnr1fYH0ygpwC0/PAL--%3E-DS-proposal?node-id=120%3A90290&t=sBqtGfm0HJBjcnmv-1)

**Comment by SOjaHPE** [comment]
Based on our past conversations, @gregfuruiye  please review this ticket and provide feedback around the why, the details and intent. Thank you.

**Comment by ericsoderberghp** [comment]
I would also like to understand why https://v2.grommet.io/tag?c=DwFQhg5gBAbmA2BXApgXgERyc9UDOAlgF5rp4C2C8uA9AHxA isn't sufficient for this need. Back when we were creating Tag we considered calling it "Label". Both seem to describe the same basic concept in English.

**Comment by SOjaHPE** [comment]
Understand, but I feel these are distinct enough to be their own classification. Tags are shaped similar to button controls and they can also be interactive based on usage. There is specific UI's for users to manage tags and applied filter tags.

Labels would be set by the system based on a defined set of values associated with the life cycle of an object type.

Currently this was marked as **low priority** with GLCP but given the inconsistencies with current implementations I'd put this as a **high priority** Design System ticket to get ahead of usage and track the UX / UI debt.

**Comment by ericsoderberghp** [comment]
They still feel semantically very similar to me as "additional information which classifies an item / element"

**Comment by gregfuruiye** [comment]
I'll attempt to be concise and direct:

1. This is a label, not a tag and not a badge
2. It should only specify additional, non-critical data about a 'thing.' Never status, never health, maybe state
3. Do NOT use color for a label beyond some neutral color. It should not imply status or state
4. The Medium article has conflicting guidance. We have 'Tags' defined and should work hard to differentiate a label from a tag (tags are interactive while labels are not)
5. 'Badge' might be identified as what we use on the filter icon or what we might use on a notification bell

I would suggest that the GLCP use cases as a whole are misleading in that the designer is exploring and not clearly identifying the strong use case need. For example, for the compatibility report check, we should NOT use a color background and status when the messaging is clearly showing a state. I saw this show up in the Wellness Dashboard and would strongly discourage this use of label.

**Comment by bettyad** [comment]
We are looking to highlight services that provide Evaluations in the GLCP homepage and the services catalogue. These highlights or labels would not be interactive but bring attention and motivate the user to click on the service card to try out the service.  We designed a mock-up and conducted tests with users and 5 out of 5 participants had difficulty identifying labels that were light gray. 
We want users to find services that provide evaluations effortlessly. Evaluations are an important part of the purchasing process —as a customer mentions they are the "bridge between issuing a PO or not" 
Ideally, in the future these highlights or labels would be smart or connected to recently provisioned or purchased services.
Please refer to this figma for use cases

https://www.figma.com/file/Lh01flLH73cFNGZiFxEFth/HPE---Office-Hours?type=design&node-id=5613-2076&mode=design&t=4OQmpnefx3zwu7UZ-4
![image](https://github.com/grommet/hpe-design-system/assets/20525083/1b7a42db-b91b-4d0f-962c-7cc76f571889)

![image](https://github.com/grommet/hpe-design-system/assets/20525083/25220067-4c7d-48f6-b5b4-12eb77074903)

![image](https://github.com/grommet/hpe-design-system/assets/20525083/16e49633-6e5e-43a7-a8ce-32c3d6e0f04f)

Use cases

- indicate an update happened/ is available
-  Supporting information on current set up/ Account type
- Differentiate between two tiles specifically type of account
- Differentiate between two tiles specifically a service with an Evaluation available

![image](https://github.com/grommet/hpe-design-system/assets/20525083/4d808e15-928b-422b-a904-20e35373e49e)

![image](https://github.com/grommet/hpe-design-system/assets/20525083/4119ed4b-b8d0-4339-8cfc-64048e53cffc)

![image](https://github.com/grommet/hpe-design-system/assets/20525083/9204f9f3-caa3-411a-9d12-321499fb1584)

**Comment by halocline** [comment]
Asked for/about again in Office Hours on June 27, 2014 by @gregfuruiye.

**Comment by SeamusLeonardHPE** [comment]
@vavalos5 if you have capacity this sprint. Also happy if we want to have a team call to hammer out some initial shape.


**Who is requesting this?** 
Experience Studio product team KOBE dependency

**What is the problem this ask is trying to solve?**
Feature request for new "label" component - non interactive element

**Add links that contain a proposal, if it exists.**
See above examples from Greg, Betty & outline from Scott

**Initial scope**

- [ ] Written documentation what it is - what it is not 

Some debate and misalignment as to how other design systems use the term - define what it mean to us

_______________________________________________________________
**What is the issue about:**

- [ ] Component

- [ ] Template

**Reviewers:**

- [ ] Reviewer 1

- [ ] Reviewer 2

**List possible consequences, involved assets and components:**
1.

**Define MVP for this ticket** (format, amount of detail, where it should live, etc):

**Follow up tickets:** (if necessary)
1.

**Checklist** (add your progress in the comments)

- [ ] Define the Problem (build a small presentation about the problem)

- [ ] Ideate (present at least two option with pros and cons)

- [ ] Build

- [ ] [Check the Component Checklist](https://hpe.atlassian.net/wiki/spaces/DS/pages/2517205174)

- [ ] Test with users

- [ ] Test with consumers

- [ ] Test with Bill for accessibility 

- [ ] [Create a Figma Branch](https://hpe.atlassian.net/wiki/spaces/DS/pages/2516288954/Branching+Merging+and+Publishing#Creating-a-New-Branch)

- [ ] Ask for review

- [ ] Publish

- [ ] Communicate

**Comment by vavalos5** [comment]
@SeamusLeonardHPE during sprint planning, we discussed this ticket was to just focus on understanding what a label is and what it is not. I see there are some deliverables to build and test for this ticket which we did not discuss. Can you please confirm that this is an exploration ticket on just identifying what a label is, gathering research and trying to get a consensus on direction with the team? Thanks!!

**Comment by SOjaHPE** [comment]
@vavalos5 thank you for asking for clarification! I can see how that template used above could cause some confusion. @SeamusLeonardHPE  can reply tomorrow if he has anything to add but to unblock you, the scope for this sprint is that bullet point under 'initial scope' -
_____________________________________________________________
**Initial scope**
- [ ] Written documentation what it is - what it is not
_____________________________________________________________
**The following statement further reinforced why we want to know, what it is & what it is not.**
_"Some debate and misalignment as to how other design systems use the term - define what it mean to us"_

**Note:** The other 'checklist' scope items in the template are there as a default, to be included as necessary depending on the need/problem to be solved.

**Comment by SeamusLeonardHPE** [comment]
@vavalos5 thanks for tagging, I updated the comment, had based it on Julias templates but realsie tehre was too much noise in there that convoluted the core deliverable.

**Comment by vavalos5** [comment]
[Figma file](https://www.figma.com/design/kr6ZBMRYipXRDkBd4o67b4/Labels?node-id=17-24926&t=hbJoydEbDfDFTF0s-1)
I've created a decision tree and decision table to differentiate when a label, tag or badge should be used. In addition, i've provided a short summary of what a label is and isn't. 

@SeamusLeonardHPE please let me know your thoughts. Looking to share out for D&D.

**Comment by vavalos5** [comment]
[Figma link](https://www.figma.com/design/kr6ZBMRYipXRDkBd4o67b4/Labels?node-id=54-8467&t=S833LLCQTSQxEHP0-1) where you'll see Tag proposals w/ variants.

The naming ‘label’ can be confusing to others as label is something that’s associated with an input.

Based on my conversation with Mike, he explained to me that tags semantically make sense for the use cases we’re trying to cover as we can remove onClick and onRemove and would still be semantically aligned.

I’ve also shared with team problems and proposals and they’ve each agreed we should steer away from using the name ‘labels’ and reserve it for what it is semantically. 

In addition, when looking at Uber (named Badge), Ant Design (named Tag) and Material (named Chip), they each take a similar approach where they have variants - it’s up to us to decide on naming.

Lastly, I’ve come across posts on Reddit and UX Exchange where other designers have come across the same problem; they’ve reached the consensus as I have that it’s up to us decide the name of the component & definition.

I’m proposing we create 1-2 variants of Tag and explain usage.

OR

we create another component named ‘chip’ or ‘pill’ which is pretty much the same thing as a tag. (not convinced on this)

@SeamusLeonardHPE Please review and let me know your thoughts. @MikeKingdom let me know if you have further inputs on this too!

**Comment by vavalos5** [comment]
I shared proposals with team and everyone seemed on board with the direction i've proposed. 

However, I wasn't fully convinced on the "marketing" tag so i'm going to withdraw that proposal as I realized we can meet this need through our CTA card. I'm including examples below i've mocked up.

![Image](https://github.com/user-attachments/assets/278df250-e57f-4132-a8f5-2fb16c93175b)

Proposal is just to add "assistive" tag to tag with background-back styling with the added content I've written on what a label is and isn't. Lastly, we should only create xs/s variants of "assistive" tag to not allow any larger sizes to be used.

Next steps would be creating those variants in Figma.

Let me know if you have any further input @SeamusLeonardHPE or if you're ok with these steps. I can close ticket and create new one for implementation in Figma.

**Comment by vavalos5** [comment]
Seamus and I connected and agreed this can be checked as completed.

Follow up ticket: #4033 to re-visit tag guidance and rebuild

**Comment by SOjaHPE** [comment]
[like]  Oja, Scott reacted to your message:
________________________________

**Comment by bettyad** [comment]
Mocking it up on the platform screen...would it look something like this? (first row of cards is new design and card on second row is old design)
![Dashboard home get started+ list view](https://github.com/user-attachments/assets/05a8c0a7-950e-40a4-a73c-11ebf083d0d7)
